### PR TITLE
Add `issue-893` Constraints

### PR DESCRIPTION
**Pull request by Gabeblis**
# Committer Notes
## Purpose
This PR introduces two constraints `component-has-non-provider-responsible-role` and `non-provider-responsible-role-references-user`, which ensures that each component dealing with leveraged systems, interconnections, or authorized services has at least one non-provider role and each non-provider responsible role references at least one user/authorized-privilege/function-performed via the "privilege-uuid" property/extension respectively. 
## Changes

Added Constraints:
- `component-has-non-provider-responsible-role`: Ensures that each component dealing with leveraged systems, interconnections, or authorized services has at least one non-provider role.

- `non-provider-responsible-role-references-user`: each non-provider responsible role references at least one user/authorized-privilege/function-performed via the "privilege-uuid" property/extension.

Test Data:
- **Invalid Test Data File**: Demonstrates the case where the fail case.
- **Valid Test Data File**: Demonstrates the case where the pass case. 

Added YAML Files
- **4 YAML Files**: Included pass and fail YAML files to validate the above constraints.

### All Submissions:

- [x] Have you selected the correct base branch per [Contributing](https://github.com/GSA/fedramp-automation/blob/master/CONTRIBUTING.md) guidance?
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/GSA/fedramp-automation/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] If applicable, have all [FedRAMP Documents Related to OSCAL Adoption](https://github.com/GSA/fedramp-automation) affected by the changes in this issue have been updated.?
- [x] If applicable, does this PR reference the issue it addresses and explain how it addresses the issue?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### CODE REVIEW

**Review by wandmagic — DISMISSED** [review]

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 1]
```diff
@@ -67,11 +67,19 @@
...
             </expect>
+            <expect id="component-has-non-provider-responsible-role" target="//component[(@type='system' and prop[@name='leveraged-authorization-uuid']) or (@type='service' and not(prop[@name='leveraged-authorization-uuid']) and prop[@name='implementation-point' and @value='external']) or (@type='interconnection') or (@type='service' and prop[@name='implementation-point' and @value='internal'] and prop[@name='direction']) or (@type='software' and prop[@name='asset-type' and @value='cli'] and prop[@name='direction'])]" test="count(responsible-role[not(@role-id='provider')]) >= 1" level="ERROR">
+                <formal-name>Component Has Non-Provider Responsible Role</formal-name>
+                <prop namespace="https://docs.oasis-open.org/sarif/sarif/v2.1.0" name="help-url" value="https://automate.fedramp.gov/documentation/ssp/4-ssp-template-to-oscal-mapping/#external-systems-and-services-not-having-fedramp-authorization"/>
+                <message>Each component dealing with leveraged systems, interconnections, or authorized services MUST have at least one responsible role other than "provider".</message>
```
[the remark this answers is not in the document]
Comment (Gabeblis) [reply src/validations/constraints/fedramp-external-constraints.xml]: Resolved: [9be50f2](https://github.com/GSA/fedramp-automation/pull/949/commits/9be50f2497f7c9d8b30d7a60347c1836691e8145) 🧙‍♂️

**Review by aj-stein-gsa — APPROVED** [review]

**Review by Rene2mt — DISMISSED** [review]
Nice work!

**Comment by Gabeblis** [comment]
@Rene2mt @aj-stein-gsa This PR just needed a rebase. Can you please give a quick re-review?

**Review by Rene2mt — APPROVED** [review]

**Review by aj-stein-gsa — APPROVED** [review]

**Review by wandmagic — APPROVED** [review]
in the future it might be good to refactor our long conditional regarding "networked" components into a variable. looks good otherwise

**Pull request merged**